packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/VariableStarViewer/VariableStarViewerContainer.spec.js

```diff
@@ -557,4 +559,34 @@ describe('Component > VariableStarViewerContainer', function () {
       }).then(done, done)
     })
   })
+
+  describe('with pan and zoom per scatter plot module', function () {
+    let wrapper
+    before(function () {
+      const subject = Factory.build('subject', {
+        locations: [
+          { 'application/json': 'http://localhost:8080/variableStar.json' },
+          { 'image/png': 'http://localhost:8080/image1.png' }
+        ]
+      })
+      wrapper = shallow(
+        <VariableStarViewerContainer
+          subject={subject}
+        />
+      )
+    })
+
+    it('should default to not have pan and zoom enabled for either scatter plot', function () {
+      expect(wrapper.state().allowPanZoom).to.equal('')
+    })
+
+    it('should set the state for which module is allowed to zoom when setAllowPanZoom is called', function () {
+      wrapper.instance().setAllowPanZoom('rawJSON')
+      expect(wrapper.state().allowPanZoom).to.equal('rawJSON')
```

Comment:
It's bad practice to test a component's internal state directly like this. Better to test the consequences of a state change instead. I'd merge this test and the following test into something like:
```js
expect(wrapper.find(VariableStarViewer).props().allowPanZoom).to.equal('rawJSON')
```

Reply:
Also, a callback which sets a prop that's then passed back down to the component that fired the callback might mean you've got components whose state is stored outside of those components. I'm not sure.

Reply:
Yes, we're following a state container and view component convention.

Reply:
> It's bad practice to test a component's internal state directly like this. Better to test the consequences of a state change instead. I'd merge this test and the following test into something like:

The props being passed down is tested in the next spec.

Reply:
```suggestion
      expect(wrapper.find(VariableStarViewer).props().allowPanZoom).to.equal('rawJSON')
```
Same comment here, so that the test is checking behaviour, not implmentation.